# PhD-Code
Code written during my PhD.

# Full PhD Bibliograpyhy

Afset, J. E., E. Anderssen, G. Bruant, J. Harel, L. Wieler, and K. Bergh. 2008. “Phylogenetic Backgrounds and Virulence Profiles of Atypical Enteropathogenic Escherichia Coli Strains from a Case-Control Study Using Multilocus Sequence Typing and DNA Microarray Analysis.” Journal of Clinical Microbiology 46 (7): 2280–90. https://doi.org/10.1128/JCM.01752-07.

Ahmed, R., C. Bopp, A. Borczyk, and S. Kasatiya. 1987. “Phage-Typing Scheme for Escherichia Coli O157:H7.” The Journal of Infectious Diseases 155 (4): 806–9. https://doi.org/10.1093/infdis/155.4.806.

Aird, D., M. G. Ross, WS. Chen, M. Danielsson, T. Fennell, C. Russ, D. Jaffe, C. Nusbaum, and A. Gnirke. 2011. “Analyzing and Minimizing PCR Amplification Bias in Illumina Sequencing Libraries.” Genome Biology 12 (2): R18–R18. https://doi.org/10.1186/gb-2011-12-2-r18.

Akashi, S., K. Joh, A. Tsuji, H. Ito, H. Hoshi, T. Hayakawa, J. Ihara, et al. 1994. “A Severe Outbreak of Haemorrhagic Colitis and Haemolytic Uraemic Syndrome Associated with Escherichia Coli 0157:H7 in Japan.” Eur J Pediatr 153. https://doi.org/10.1007/BF02190685.

Altschul, S. F., W. Gish, W. Miller, E. W. Myers, and D. J. Lipman. 1990. “Basic Local Alignment Search Tool.” Journal of Molecular Biology 215 (3): 403–10. https://doi.org/10.1016/S0022-2836(05)80360-2.

Ardui, S., A. Ameur, J. R. Vermeesch, and M. S. Hestand. 2018. “Single Molecule Real-Time (SMRT) Sequencing Comes of Age: Applications and Utilities for Medical Diagnostics.” Nucleic Acids Research 46 (5): 2159–68. https://doi.org/10.1093/nar/gky066.

Armstrong, . L., J. Hollingsworth, and J. G. Morris. 1996. “Emerging Foodborne Pathogens: Escherichia Coli O157:H7 as a Model of Entry of a New Pathogen into the Food Supply of the Developed World.” Epidemiologic Reviews 18 (1): 29–51. https://doi.org/10.1093/oxfordjournals.epirev.a017914.

Arndt, D., J. R. Grant, A. Marcu, T. Sajed, A. Pon, Y. Liang, and D. S. Wishart. 2016. “PHASTER: A Better, Faster Version of the PHAST Phage Search Tool.” Nucleic Acids Research 44 (W1): W16–21. https://doi.org/10.1093/nar/gkw387.

Asadulghani, M., Y. Ogura, T. Ooka, T. Itoh, A. Sawaguchi, A. Iguchi, K. Nakayama, and T. Hayashi. 2009. “The Defective Prophage Pool of Escherichia Coli O157: Prophage-Prophage Interactions Potentiate Horizontal Transfer of Virulence Determinants.” PLoS Pathog 5 (5): e1000408. https://doi.org/10.1371/journal.ppat.1000408.

Ashton, P. M., N. Perry, R. Ellis, L. Petrovska, J. Wain, K. A. Grant, C. Jenkins, and T. J. Dallman. 2015. “Insight into Shiga Toxin Genes Encoded by Escherichia Coli O157 from Whole Genome Sequencing.” PeerJ 3: e739. https://doi.org/10.7717/peerj.739.

Balasubramanian, S., M.S. Osburne, H. BrinJones, A. K. Tai, and J. M. Leong. 2019. “Prophage Induction, but Not Production of Phage Particles, Is Required for Lethal Disease in a Microbiome-Replete Murine Model of Enterohemorrhagic E. Coli Infection.” PLOS Pathogens 15 (1): e1007494. https://doi.org/10.1371/journal.ppat.1007494.

Barquist, L., M. Mayho, C. Cummins, A. K. Cain, C. J. Boinett, A. J. Page, G. C. Langridge, M. A. Quail, J. A. Keane, and J. Parkhill. 2016. “The TraDIS Toolkit: Sequencing and Analysis for Dense Transposon Mutant Libraries.” Bioinformatics (Oxford, England) 32 (7): 1109–11. https://doi.org/10.1093/bioinformatics/btw022.

Belén, Ana, Ibarz Pavón, and Martin C J Maiden. 2009. “Molecular Sequence Typing.” Methods in Molecular Biology 551 (8): 129–40. https://doi.org/10.1007/978-1-60327-999-4.

Betz, J., M. Bielaszewska, A. Thies, HU. Humpf, K. Dreisewerd, H. Karch, K. S. Kim, A. W. Friedrich, and J. Müthing. 2011. “Shiga Toxin Glycosphingolipid Receptors in Microvascular and Macrovascular Endothelial Cells: Differential Association with Membrane Lipid Raft Microdomains.” Journal of Lipid Research 52 (4): 618–34. https://doi.org/10.1194/jlr.M010819.

Blanco, M., J. E. Blanco, J. Blanco, E. A. Gonzalez, A. Mora, C. Prado, L. Fernandez, M. Rio, J. Ramos, and M. P. Alonso. 1996. “Prevalence and Characteristics of Escherichia Coil Serotype O157:H7 and Other Verotoxin-Producing E. Coli in Healthy Cattle.” Epidemiology and Infection 117 (2): 251–57. https://doi.org/DOI: 10.1017/S0950268800001424.

Byrne, L., R. Elson, T. J. Dallman, N. Perry, P. Ashton, J. Wain, G. K. Adak, K. A. Grant, and C. Jenkins. 2014. “Evaluating the Use of Multilocus Variable Number Tandem Repeat Analysis of Shiga Toxin-Producing Escherichia Coli O157 as a Routine Public Health Tool in England.” PLoS ONE 9 (1). https://doi.org/10.1371/journal.pone.0085901.

Byrne, L., C. Jenkins, N. Launders, R. Elson, and G. K. Adak. 2015. “The Epidemiology, Microbiology and Clinical Impact of Shiga Toxin-Producing Escherichia Coli in England, 2009–2012.” Epidemiology and Infection 143 (16): 3475–87. https://doi.org/DOI: 10.1017/S0950268815000746.

Camacho, C., G. Coulouris, V. Avagyan, N. Ma, J. Papadopoulos, K. Bealer, and T. L. Madden. 2009. “BLAST+: Architecture and Applications.” BMC Bioinformatics 10: 421. https://doi.org/10.1186/1471-2105-10-421.

Chase-Topping, Margo, David Gally, Chris Low, Louise Matthews, and Mark Woolhouse. 2008. “Super-Shedding and the Link between Human Infection and Livestock Carriage of Escherichia Coli O157.” Nature Reviews. Microbiology 6 (12): 904–12. https://doi.org/10.1038/nrmicro2029.

Chin, C. S., D. Alexander, P. Marks, A. Klammer, J. Drake, C. Heiner, A. Clum, et al. 2013. “Nonhybrid, Finished Microbial Genome Assemblies from Long-Read SMRT Sequencing Data.” Nature Methods 10 (6). https://doi.org/10.1038/nmeth.2474.

Cowley, L. A., T. J. Dallman, S. Fitzgerald, N. Irvine, P. J. Rooney, S. P. McAteer, M. Day, et al. 2016. “Short-Term Evolution of Shiga Toxin-Producing Escherichia Coli O157:H7 between Two Food-Borne Outbreaks.” Microbial Genomics 2 (9): e000084. https://doi.org/10.1099/mgen.0.000084.

Cunningham, S. A., N. Chia, P. R. Jeraldo, D. J. Quest, J. A. Johnson, D. J. Boxrud, A. J. Taylor, et al. 2017. “Comparison of Whole-Genome Sequencing Methods for Analysis of Three Methicillin-Resistant Staphylococcus Aureus Outbreaks.” Journal of Clinical Microbiology 55 (6): 1946–53. https://doi.org/10.1128/JCM.00029-17.

Dallman, T. J., P. M. Ashton, L. Byrne, N. Perry, L. Petrovska, R. Ellis, L. Allison, et al. 2015. “Applying Phylogenomics to Understand the Emergence of Shiga-Toxin-Producing Escherichia Coli O157:H7 Strains Causing Severe Human Disease in the UK.” Microbial Genomics. https://doi.org/10.1099/mgen.0.000029.

Dallman, T. J., P. Ashton, U. Schafer, A. Jironkin, A. Painset, S. Shaaban, H. Hartman, et al. 2018. “SnapperDB: A Database Solution for Routine Sequencing Analysis of Bacterial Isolates.” Bioinformatics 34 (17): 3028–29. https://doi.org/10.1093/bioinformatics/bty212.

Darling, A. E., B. Mau, and N. T. Perna. 2010. “Progressivemauve: Multiple Genome Alignment with Gene Gain, Loss and Rearrangement.” PLoS ONE 5 (6). https://doi.org/10.1371/journal.pone.0011147.

Darling, A. E., I. Miklós, and M. A. Ragan. 2008. “Dynamics of Genome Rearrangement in Bacterial Populations.” PLOS Genetics 4 (7): e1000128. https://doi.org/10.1371/journal.pgen.1000128.

Delcher, A. L., K. A. Bratke, E. C. Powers, and S. L. Salzberg. 2007. “Identifying Bacterial Genes and Endosymbiont DNA with Glimmer.” Bioinformatics 23 (6): 673–79. https://doi.org/10.1093/bioinformatics/btm009.

Eppinger, M., M. K. Mammel, J. E. Leclerc, J. Ravel, and T. A. Cebula. 2011. “Genomic Anatomy of Escherichia Coli O157:H7 Outbreaks.” Proceedings of the National Academy of Sciences 108 (50): 20142–47. https://doi.org/10.1073/pnas.1107176108.

Feng, P., and S. Monday. 2000. “Multiplex PCR for Detection of Trait and Virulence Factors in Enterohemorrhagic Escherichia Coli Serotypes.” Molecular and Cellular Probes 14 (6): 333–37. https://doi.org/10.1006/mcpr.2000.0323.

Fijalkowska, I. J., R. M. Schaaper, and P. Jonczyk. 2012. “DNA Replication Fidelity in Escherichia Coli: A Multi-DNA Polymerase Affair.” FEMS Microbiology Reviews 36 (6): 1105–21. https://doi.org/10.1111/j.1574-6976.2012.00338.x.

Fitzgerald, S. F., A. E. Beckett, J. Palarea-Albaladejo, S. McAteer, S. Shaaban, J. Morgan, N. I. Ahmad, et al. 2019. “Shiga Toxin Sub-Type 2a Increases the Efficiency of Escherichia Coli O157 Transmission between Animals and Restricts Epithelial Regeneration in Bovine Enteroids.” PLoS Pathogens 15 (10). https://doi.org/10.1371/journal.ppat.1008003.

Fratamico, P. M., S. K. Sackitey, M. Wiedmann, and Y. D. Ming. 1995. “Detection of Escherichia Coli O157:H7 by Multiplex PCR.” Journal of Clinical Microbiology 33 (8): 2188–91.
Fuller, C. A., C. A. Pellino, M. J. Flagler, J. E. Strasser, and A. A. Weiss. 2011. “Shiga Toxin Subtypes Display Dramatic Differences in Potency.” Edited by S R Blanke. Infection and Immunity 79 (3): 1329 LP – 1337. https://doi.org/10.1128/IAI.01182-10.

Geneious. 2019. “Geneious.” 2019. https://www.geneious.com/. Accessed 01/09/2019. Web Archive: https://web.archive.org/web/20190906041414/https://www.geneious.com/.

González-Escalona, N., M. A. Allard, E. W. Brown, S. Sharma, and M. Hoffmann. 2019. “Nanopore Sequencing for Fast Determination of Plasmids, Phages, Virulence Markers, and Antimicrobial Resistance Genes in Shiga Toxin-Producing Escherichia Coli.” Plos One 14 (7): e0220494. https://doi.org/10.1371/journal.pone.0220494.

Greig, D. R., C. Jenkins, S. Gharbia, and T. J. Dallman. 2019. “Comparison of Single-Nucleotide Variants Identified by Illumina and Oxford Nanopore Technologies in the Context of a Potential Outbreak of Shiga Toxin–Producing Escherichia Coli.” GigaScience 8 (8). https://doi.org/10.1093/gigascience/giz104.

Hänninen, M. L., M. Hakkinen, and H. Rautelin. 1999. “Stability of Related Human and Chicken Campylobacter Jejuni Genotypes after Passage through Chick Intestine Studied by Pulsed-Field Gel Electrophoresis.” Applied and Environmental Microbiology 65 (5): 2272–75. https://www.ncbi.nlm.nih.gov/pubmed/10224037.

Ho, N. K., A. C. Henry, K. Johnson-Henry, and P.M. Sherman. 2013. “Pathogenicity, Host Responses and Implications for Management of Enterohemorrhagic Escherichia Coli O157:H7 Infection.” Canadian Journal of Gastroenterology = Journal Canadien de Gastroenterologie 27 (5): 281–85. https://doi.org/10.1155/2013/138673.

Holmes, A., T. J. Dallman, S. Shabaan, M. Hanson, and L. Allison. 2018. “Validation of Whole-Genome Sequencing for Identification and Characterization of Shiga Toxin-Producing Escherichia Coli To Produce Standardized Data To Enable Data Sharing.” Edited by Alexander Mellmann. Journal of Clinical Microbiology 56 (3): e01388-17. https://doi.org/10.1128/JCM.01388-17.

Hyatt, D., GL. Chen, P. F. Locascio, M. L. Land, F. W. Larimer, and L. J. Hauser. 2010. “Prodigal: Prokaryotic Gene Recognition and Translation Initiation Site Identification.” BMC Bioinformatics 11 (March): 119. https://doi.org/10.1186/1471-2105-11-119.

Illumina, Inc. 2019. “Illumina Platforms.” 2019. https://www.illumina.com/systems/sequencing-platforms.html. Accessed 31/08/2019. Web Archive: https://web.archive.org/web/20191006190049/https://www.illumina.com/systems/sequencing-platforms.html.

Imamovic, L., R. Tozzoli, V. Michelacci, F. Minelli, M. L. Marziano, A. Caprioli, and S. Morabito. 2010. “OI-57, a Genomic Island of Escherichia Coli O157, Is Present in Other Seropathotypes of Shiga Toxin-Producing E. Coli Associated with Severe Human Disease.” Infection and Immunity 78 (11): 4697 LP – 4704. https://doi.org/10.1128/IAI.00512-10.

Inkscape. 2019. “InkScape.” 2019. https://inkscape.org/. Accessed 31/08/2019. Web Archive: https://web.archive.org/web/20190915014942/https://inkscape.org/.

Izumiya, H., Y. Pei, J. Terajima, M. Ohnishi, T. Hayashi, S. Iyoda, and H. Watanabe. 2010. “New System for Multilocus Variable-Number Tandem-Repeat Analysis of the Enterohemorrhagic Escherichia Coli Strains Belonging to Three Major Serogroups: O157, O26, and O111.” Microbiology and Immunology 54 (10): 569–77. https://doi.org/10.1111/j.1348-0421.2010.00252.x.

Izumiya, H., J. Terajima, A. Wada, Y. Inagaki, K. I. Itoh, K. Tamura, and H. Watanabe. 1997. “Molecular Typing of Enterohemorrhagic Escherichia Coli O157:H7 Isolates in Japan by Using Pulsed-Field Gel Electrophoresis.” Journal of Clinical Microbiology 35 (7): 1675–80.

Janowicz, A., F. De Massis, M. Ancora, C. Cammà, C. Patavino, A. Battisti, K. Prior, et al. 2018. “Core Genome Multilocus Sequence Typing and Single Nucleotide Polymorphism Analysis in the Epidemiology of Brucella Melitensis Infections.” Edited by Daniel J Diekema. Journal of Clinical Microbiology 56 (9): e00517-18. https://doi.org/10.1128/JCM.00517-18.

Kaguni, J.M. 2011. “Replication Initiation at the Escherichia Coli Chromosomal Origin.” Current Opinion in Chemical Biology 15 (5): 606–13. https://doi.org/10.1016/j.cbpa.2011.07.016.

Kaper, J., J. Nataro, and H. Mobley. 2004. “Pathogenic Escherichia Coli.” Nat Rev Microbiol 2.

Karch, H., E. Denamur, U. Dobrindt, B. B. Finlay, R. Hengge, L. Johannes, E. Z. Ron, T. Tønjum, P. J. Sansonetti, and M. Vicente. 2012. “The Enemy within Us: Lessons from the 2011 European Escherichia Coli O104:H4 Outbreak.” EMBO Molecular Medicine 4 (9): 841–48. https://doi.org/10.1002/emmm.201201662.

Katani, R., R. Cote, J. A. R. Garay, L. Li, T. M. Arthur, C. DebRoy, M. M. Mwangi, and V. Kapur. 2016. “Complete Genome Sequence of SS52, a Strain of Escherichia Coli O157: H7 Recovered from Supershedder Cattle.” Genome Announcements 3 (2): 1999–2000. https://doi.org/10.1128/genomeA.01569-14.

Keim, P., L. B. Price, A. M. Klevytska, K. L. Smith, J. M. Schupp, R. Okinaka, P. J. Jackson, and M. E. Hugh-Jones. 2000. “Multiple-Locus Variable-Number Tandem Repeat Analysis Reveals Genetic Relationships within Bacillus Anthracis.” Journal of Bacteriology 182 (10): 2928–36. https://doi.org/10.1128/JB.182.10.2928-2936.2000.

Khakhria, R., D. Duck, and H. Lior. 1990. “Extended Phage-Typing Scheme for Escherichia Coli 0157:H7.” Epidemiology and Infection 105 (3): 511–20. https://doi.org/10.1017/S0950268800048135.

Koren, S., B. P. Walenz, K. Berlin, J. R. Miller, N. H. Bergman, and A. M. Phillippy. 2017. “Canu: Scalable and Accurate Long-Read Assembly via Adaptive k-Mer Weighting and Repeat Separation.” Genome Research 27 (5): 722–36. https://doi.org/10.1101/gr.215087.116.

Krzywinski, M.I., J. E. Schein, I. Birol, J. Connors, R. Gascoyne, D. Horsman, S. J. Jones, and M. A. Marra. 2009. “Circos: An Information Aesthetic for Comparative Genomics.” Genome Research, June. https://doi.org/10.1101/gr.092759.109.

Kulasekara, B. R., M. Jacobs, Y. Zhou, Z. Wu, E. Sims, C. Saenphimmachak, L. Rohmer, et al. 2009. “Analysis of the Genome of the Escherichia Coli O157:H7 2006 Spinach-Associated Outbreak Isolate Indicates Candidate Genes That May Enhance Virulence.” Infection and Immunity 77 (9): 3713–21. https://doi.org/10.1128/IAI.00198-09.

Laing, C. R., Y. Zhang, M.W. Gilmour, V. Allen, R. Johnson, J. E. Thomas, and V. P. J. Gannon. 2012. “A Comparison of Shiga-Toxin 2 Bacteriophage from Classical Enterohemorrhagic Escherichia Coli Serotypes and the German E. Coli O104:H4 Outbreak Strain.” PLoS ONE 7 (5). https://doi.org/10.1371/journal.pone.0037362.

Lee, H., T.G. Doak, E. Popodi, P. L. Foster, and H. Tang. 2016. “Insertion Sequence-Caused Large-Scale Rearrangements in the Genome of Escherichia Coli.” Nucleic Acids Research 44 (15): 7109–19. https://doi.org/10.1093/nar/gkw647.

Li, H., and R. Durbin. 2009. “Fast and Accurate Short Read Alignment with Burrows-Wheeler Transform.” Bioinformatics 25 (14): 1754–60. https://doi.org/10.1093/bioinformatics/btp324.

Lim, J.Y., J. Yoon, and C. J. Hovde. 2010. “A Brief Overview of Escherichia Coli O157:H7 and Its Plasmid O157.” Journal of Microbiology and Biotechnology 20 (1): 5–14. https://www.ncbi.nlm.nih.gov/pubmed/20134227.

Luo, H., CT. Zhang, and F. Gao. 2014. “Ori-Finder 2, an Integrated Tool to Predict Replication Origins in the Archaeal Genomes.” Frontiers in Microbiology 5: 482. https://doi.org/10.3389/fmicb.2014.00482.

Maharjan, R. P., and T. Ferenci. 2018. “The Impact of Growth Rate and Environmental Factors on Mutation Rates and Spectra in Escherichia Coli.” Environmental Microbiology Reports 10 (6): 626–33. https://doi.org/10.1111/1758-2229.12661.

Maiden, M. C. J. 2006. “Multilocus Sequence Typing of Bacteria.” Annual Review of Microbiology 60 (1): 561–88. https://doi.org/10.1146/annurev.micro.59.030804.121325.

Maiden, M. C. J., J. A. Bygraves, E. Feil, G. Morelli, J. E. Russell, R. Urwin, Q. Zhang, et al. 1998. “Multilocus Sequence Typing: A Portable Approach to the Identification of Clones within Populations of Pathogenic Microorganisms (Molecular TypingNeisseria Meningitidishousekeeping GenesWorld-Wide Webhyper-Virulent Clones).” Proc. Natl. Acad. Sci. USA 95 (March): 3140–45. www.pnas.org.

Matthews, T. D., and S. Maloy. 2010. “Fitness Effects of Replichore Imbalance in Salmonella Enterica.” Journal of Bacteriology 192 (22): 6086 LP – 6088. https://doi.org/10.1128/JB.00649-10.

Mcdaniel, T. K., K. G. Jarvis, M. S. Donnenberg, and J. B. Kaper. 1995. “A Genetic Locus of Enterocyte Effacement Conserved among Diverse Enterobacterial Pathogens.” Proceedings of the National Academy of Sciences of the United States of America 92 (5): 1664–68. https://doi.org/10.1073/pnas.92.5.1664.

McKenna, A., M. Hanna, E. Banks, A. Sivachenko, K. Cibulskis, A. Kernytsky, K. Garimella, et al. 2010. “The Genome Analysis Toolkit: A MapReduce Framework for Analyzing next-Generation DNA Sequencing Data.” Genome Res 20 (9): 1297–1303. https://doi.org/10.1101/gr.107524.110.

Melton-Celsa, A. R. 2014. “Shiga Toxin (Stx) Classification, Structure, and Function.” Microbiology Spectrum 2 (4): 10.1128/microbiolspec.EHEC-0024-2013–2013. https://doi.org/10.1128/microbiolspec.EHEC-0024-2013.

Mikheyev, A. S., and M. M. Tin. 2014. “A First Look at the Oxford Nanopore MinION Sequencer.” Mol Ecol Resour 14 (6): 1097–1102. https://doi.org/10.1111/1755-0998.12324.

Nadon, C. A., E. Trees, L. K. Ng, E. Møller Nielsen, A. Reimer, N. Maxwell, K. A. Kubota, P. Gerner-Smidt, and MLVA Harmonization Working Group. 2013. “Development and Application of MLVA Methods as a Tool for Inter-Laboratory Surveillance.” Euro Surveillance : Bulletin Europeen Sur Les Maladies Transmissibles = European Communicable Disease Bulletin 18 (35): 20565. https://doi.org/10.2807/1560-7917.es2013.18.35.20565.

Needleman, S. B., and C. D. Wunsch. 1970. “A General Method Applicable to the Search for Similarities in the Amino Acid Sequence of Two Proteins.” Journal of Molecular Biology 48 (3): 443–53. https://doi.org/https://doi.org/10.1016/0022-2836(70)90057-4.

Neylon, C., A. V. Kralicek, T. M. Hill, and N. E. Dixon. 2005. “Replication Termination in Escherichia Coli: Structure and Antihelicase Activity of the Tus-Ter Complex.” Microbiology and Molecular Biology Reviews : MMBR 69 (3): 501–26. https://doi.org/10.1128/MMBR.69.3.501-526.2005.

O’Brien, A. D., J. W. Newland, S. F. Miller, R. K. Holmes, H. W. Smith, and S. B. Formal. 1984. “Shiga-like Toxin-Converting Phages from Escherichia Coli Strains That Cause Hemorrhagic Colitis or Infantile Diarrhea.” Science 226 (4675): 694 LP – 696. https://doi.org/10.1126/science.6387911.

Ofir, G., and R. Sorek. 2018. “Contemporary Phage Biology: From Classic Models to New Insights.” Cell 172 (6): 1260–70. https://doi.org/10.1016/j.cell.2017.10.045.

Ogura, Y., S. I. Mondal, M. R. Islam, T. Mako, K. Arisawa, K. Katsura, T. Ooka, et al. 2015. “The Shiga Toxin 2 Production Level in Enterohemorrhagic Escherichia Coli O157:H7 Is Correlated with the Subtypes of Toxin-Encoding Phage.” Sci Rep 5: 16663. https://doi.org/10.1038/srep16663.

Ohnishi, M, K Kurokawa, and T Hayashi. 2001. “Diversification of Escherichia Coli Genomes: Are Bacteriophages the Major Contributors?” Trends in Microbiology 9 (10).

Ooka, T., Y. Ogura, M. Asadulghani, M. Ohnishi, K. Nakayama, J. Terajima, H. Watanabe, and T. Hayashi. 2009. “Inference of the Impact of Insertion Sequence (IS) Elements on Bacterial Genome Diversification through Analysis of Small-Size Structural Polymorphisms in Escherichia Coli O157 Genomes.” Genome Research 19. https://doi.org/10.1101/gr.089615.108.

Oxford Nanopore. 2017. “Nanopore 1Mbp Read.” 2017. https://nanoporetech.com/about-us/news/world-first-continuous-dna-sequence-more-million-bases-achieved-nanopore-sequencing. Accessed 31/08/2019. Web Archive: NA.

———. 2019. “Oxford Nanopore.” 2019. https://nanoporetech.com/. Accessed 31/08/2019. Web Archive: https://web.archive.org/web/20190902104823/https://nanoporetech.com/.
Pacheco, A. R., and V. Sperandio. 2012. “Shiga Toxin in Enterohemorrhagic E.Coli: Regulation and Novel Anti-Virulence Strategies.” Frontiers in Cellular and Infection Microbiology 2 (June): 81. https://doi.org/10.3389/fcimb.2012.00081.

Pacific Biosciences. 2019. “PacBio.” 2019. https://www.pacb.com/. Accessed: 31/08/2019. Web Archive: https://web.archive.org/web/20190905232920/https://www.pacb.com/.

Pearce, M. E., NF. Alikhan, T. J. Dallman, Z. Zhou, K. Grant, and M. C. J. Maiden. 2018. “Comparative Analysis of Core Genome MLST and SNP Typing within a European Salmonella Serovar Enteritidis Outbreak.” International Journal of Food Microbiology 274 (June): 1–11. https://doi.org/10.1016/j.ijfoodmicro.2018.02.023.

Pei, Y., J. Terajima, Y. Saito, R. Suzuki, N. Takai, H. Izumiya, T. Morita-Ishihara, et al. 2008. “Molecular Characterization of Enterohemorrhagic Escherichia Coli O157:H7 Isolates Dispersed across Japan by Pulsed-Field Gel Electrophoresis and Multiple-Locus Variable-Number Tandem Repeat Analysis.” Japanese Journal of Infectious Diseases 61 (1): 58–64.

Perna, N., G. Plunkett III, V. Burland, B. Mau, J. Glasner, D. Rose, G. Mayhew, et al. 2001. “Genome Sequence of Enterohaemorrhagic Escherichia Coli O157:H7.” Nature 409. https://doi.org/10.1038/35054089.

Persson, S., K. E. P. Olsen, S. Ethelberg, and F. Scheutz. 2007. “Subtyping Method for Escherichia Coli Shiga Toxin (Verocytotoxin) 2 Variants and Correlations to Clinical Manifestations.” Journal of Clinical Microbiology 45 (6): 2020–24. https://doi.org/10.1128/JCM.02591-06.

PHE Bioinformatics Unit. 2015. “PHEnix.” 2015. https://github.com/phe-bioinformatics/PHEnix. Accessed 31/08/2019. Web Archive: NA.

Pruimboom-Brees, I. M., T. W. Morgan, M. R. Ackermann, E. D. Nystrom, J. E. Samuel, N. A. Cornick, and H. W. Moon. 2000. “Cattle Lack Vascular Receptors for Escherichia Coli O157:H7 Shiga Toxins.” Proceedings of the National Academy of Sciences of the United States of America 97 (19): 10325–29. https://doi.org/10.1073/pnas.190329997.

Quick, J., N. D. Grubaugh, S.T. Pullan, I. M. Claro, A. D. Smith, K. Gangavarapu, G. Oliveira, et al. 2017. “Multiplex PCR Method for MinION and Illumina Sequencing of Zika and Other Virus Genomes Directly from Clinical Samples.” Nature Protocols 12 (May): 1261. https://doi.org/10.1038/nprot.2017.066.

Quick, J., N. J. Loman, S. Duraffour, J. T. Simpson, E. Severi, L. Cowley, J. A. Bore, et al. 2016. “Real-Time, Portable Genome Sequencing for Ebola Surveillance.” Nature 530 (February): 228. https://doi.org/10.1038/nature16996.

R Core Team. 2019. “R: A Language and Environment for Statistical Computing.” 2019. https://www.r-project.org/. Accessed: 01/09/2019. Web Archive: https://web.archive.org/web/20190831232313/http://www.r-project.org/.

Raeside, C., J. Gaffé, D. E. Deatherage, O. Tenaillon, A. M. Briska, R. N. Ptashkin, S. Cruveiller, et al. 2014. “Large Chromosomal Rearrangements during a Long-Term Evolution Experiment with Escherichia Coli.” Edited by Søren Baquero Molin Fernando. MBio 5 (5): e01377-14. https://doi.org/10.1128/mBio.01377-14.

Ratnam, S., S. B. March, R. Ahmed, and G.S. Bezanson. 1988. “Characterization of Escherichia Coli Serotype 0157 : H7.” Journal of Clinical Microbiology 26 (10): 2006–12.
Ravindran, P., and A. Gupta. 2015. “Image Processing for Optical Mapping.” GigaScience 4 (November): 57. https://doi.org/10.1186/s13742-015-0096-z.

Reid, S. D., C. J. Herbelin, A. C. Bumbaugh, R. K. Selander, and T. S. Whittam. 2000. “Parallel Evolution of Virulence in Pathogenic Escherichia Coli.” Nature 406 (6791): 64–67. https://doi.org/10.1038/35017546.

Ribot, E. M., M. A. Fair, R. Gautom, D. N. Cameron, S. B. Hunter, B. Swaminathan, and T. J. Barrett. 2006. “Standardization of Pulsed-Field Gel Electrophoresis Protocols for the Subtyping of Escherichia Coli O157:H7, Salmonella, and Shigella for PulseNet.” Foodborne Pathogens and Disease 3 (1): 59–67. https://doi.org/10.1089/fpd.2006.3.59.

Rocha, E. P. C. 2004. “The Replication-Related Organization of Bacterial Genomes.” Microbiology 150 (6): 1609–27. https://doi.org/10.1099/mic.0.26974-0.

Saile, N., A. Voigt, S. Kessler, T. Stressler, J. Klumpp, L. Fischer, and H. Schmidt. 2016. “Escherichia Coli O157:H7 Strain EDL933 Harbors Multiple Functional Prophage-Associated Genes Necessary for the Utilization of 5-N-Acetyl-9-O-Acetyl Neuraminic Acid as a Growth Substrate.” Applied and Environmental Microbiology 82 (19): 5940–50. https://doi.org/10.1128/AEM.01671-16.

Scheutz, F., L. D. Teel, L. Beutin, D. Pierard, G. Buvens, H. Karch, A. Mellmann, et al. 2012. “Multicenter Evaluation of a Sequence-Based Protocol for Subtyping Shiga Toxins and Standardizing Stx Nomenclature.” J Clin Microbiol 50 (9): 2951–63. https://doi.org/10.1128/JCM.00860-12.

Schmidt, H., and M. Hensel. 2004. “Pathogenicity Islands in Bacterial Pathogenesis.” Clinical Microbiology Reviews 17 (1): 14–56. https://doi.org/10.1128/cmr.17.1.14-56.2004.

Schmidt, S., A. Gerasimova, F. A. Kondrashov, I. A. Adzuhbei, A. S. Kondrashov, and S. Sunyaev. 2008. “Hypermutable Non-Synonymous Sites Are under Stronger Negative Selection.” PLOS Genetics 4 (11): e1000281. https://doi.org/10.1371/journal.pgen.1000281.

Schutz, K., L. A. Cowley, S. Shaaban, A. Carroll, E. McNamara, D. L. Gally, G. Godbole, C. Jenkins, and T. J. Dallman. 2017. “Evolutionary Context of Non-Sorbitol-Fermenting Shiga Toxin-Producing Escherichia Coli O55:H7.” Emerging Infectious Diseases 23 (12): 1966–73. https://doi.org/10.3201/eid2312.170628.

Seemann, T. 2014. “Prokka: Rapid Prokaryotic Genome Annotation.” Bioinformatics 30 (14): 2068–69. https://doi.org/10.1093/bioinformatics/btu153.

Shaaban, S., L. A. Cowley, S. P. McAteer, C. Jenkins, T. J. Dallman, J. L. Bono, and D. L. Gally. 2016. “Evolution of a Zoonotic Pathogen: Investigating Prophage Diversity in Enterohaemorrhagic Escherichia Coli O157 by Long-Read Sequencing.” Microbial Genomics 2 (12). https://doi.org/10.1099/mgen.0.000096.

Sharp, P. M., D. C. Shields, K. H. Wolfe, and W. H. Li. 1989. “Chromosomal Location and Evolutionary Rate Variation in Enterobacterial Genes.” Science 246 (4931): 808 LP – 810. https://doi.org/10.1126/science.2683084.

Smith, T. F., and M. S. Waterman. 1981. “Identification of Common Molecular Subsequences.” Journal of Molecular Biology 147 (1): 195–97. https://doi.org/10.1016/0022-2836(81)90087-5.

Stewart, A. C., B. Osborne, and T. D. Read. 2009. “DIYA: A Bacterial Annotation Pipeline for Any Genomics Lab.” Bioinformatics 25 (7): 962–63. https://doi.org/10.1093/bioinformatics/btp097.

Stothard, P. 2000. “The Sequence Manipulation Suite: JavaScript Programs for Analyzing and Formatting Protein and DNA Sequences.” BioTechniques 28 (6): 1102–4. https://doi.org/10.2144/00286ir01.

Sullivan, M. J., N. K. Petty, and S. A. Beatson. 2011. “Easyfig: A Genome Comparison Visualiser.” Bioinformatics 27 (7): 1009–10. https://doi.org/10.1093/bioinformatics/btr039.

Tobe, T., S. A. Beatson, H. Taniguchi, H. Abe, C. M. Bailey, A. Fivian, R. Younis, et al. 2006. “An Extensive Repertoire of Type III Secretion Effectors in Escherichia Coli O157 and the Role of Lambdoid Phages in Their Dissemination.” Proceedings of the National Academy of Sciences of the United States of America 103 (40): 14941–46.

Underwood, A. P., T. Dallman, N. R. Thomson, M. Williams, K. Harker, N. Perry, B. Adak, et al. 2013. “Public Health Value of Next-Generation DNA Sequencing of Enterohemorrhagic Escherichia Coli Isolates from an Outbreak.” Journal of Clinical Microbiology 51 (1): 232–37. https://doi.org/10.1128/JCM.01696-12.

Wickham, H. 2009. Ggplot2: Elegant Graphics for Data Analysis. Springer-Verlag New York.

Xiao, J., Z. Zhang, J. Wu, and J. Yu. 2015. “A Brief Review of Software Tools for Pangenomics.” Genomics, Proteomics & Bioinformatics 13 (1): 73–76. https://doi.org/10.1016/j.gpb.2015.01.007.

Xu, X., S. P. McAteer, J.J. Tree, D. J. Shaw, E. B. K. Wolfson, S. A. Beatson, A. J. Roe, et al. 2012. “Lysogeny with Shiga Toxin 2-Encoding Bacteriophages Represses Type III Secretion in Enterohemorrhagic Escherichia Coli.” PLoS Pathogens 8 (5): e1002672. https://doi.org/10.1371/journal.ppat.1002672.

Zhou, Y., Y. Liang, K. H. Lynch, J. J. Dennis, and D. S. Wishart. 2011. “PHAST: A Fast Phage Search Tool.” Nucleic Acids Research 39 (SUPPL. 2): 1–6. https://doi.org/10.1093/nar/gkr485.
